llvm/include/llvm/Support/SYCLPropertySetIO.h

```diff
@@ -66,8 +66,13 @@ class SYCLPropertyValue {
     return static_cast<Type>(T);
   }
 
-  ~SYCLPropertyValue() {}
-
+  ~SYCLPropertyValue() {
+    if (std::holds_alternative<std::byte *>(Val)) {
+      auto ByteArrayVal = std::get<std::byte *>(Val);
+      if (ByteArrayVal)
+        delete[] ByteArrayVal;
```

Comment:
Please just use unique_ptr

Reply:
Hi @arsenm 

I will give it another shot. I had a bit of an issue in getting unique_ptr to work with remainder of the code. I must be missing something. I will put up a change soon. Thanks again for looking into this.

Sincerely

Reply:
Hi @arsenm 

I added use of unique_ptr in 0d5af17a958984fba156a95e6adf7a5b6879c1f2
It was a good learning curve for me.

Thanks
Sincerely